typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -190,13 +201,20 @@ async function main(address: string, options: { output: string }): Promise<void>
           log("received message for unknown subscription %s", event.subscriptionId);
           return;
         }
-        void writer.addMessage({
-          channelId: subscription.mcapChannelId,
-          sequence: subscription.messageCount++,
-          logTime: BigInt(Date.now()) * 1_000_000n,
-          publishTime: event.timestamp,
-          data: new Uint8Array(event.data.buffer, event.data.byteOffset, event.data.byteLength),
-        });
+
+        writeMsgQueue
+          .add(async () => {
+            await writer.addMessage({
+              channelId: subscription.mcapChannelId,
+              sequence: subscription.messageCount++,
+              logTime: BigInt(Date.now()) * 1_000_000n,
+              publishTime: event.timestamp,
+              data: new Uint8Array(event.data.buffer, event.data.byteOffset, event.data.byteLength),
+            });
+          })
+          .catch(() => {
+            log(`Dropping message due to queue limit (${options.queueLimit}) reached.`);
```

Comment:
would this also be triggered if the underlying addMessage resulted in an error? Shouldn't really matter in practice I guess

Reply:
Looks like it, judging from the documentation of `PromiseQueue.add`. I think I will change it to just log the returned error here

> @returns
> A promise that forwards the resolution/rejection of the promise returned by promiseGenerator, or immediately rejects if maxQueuedPromise is exceeded.